wcfsetup/install/files/lib/system/session/SessionHandler.class.php

```diff
@@ -405,46 +370,114 @@ public function getUser() {
 	}
 	
 	/**
-	 * Tries to read existing session identified by the given session id.
-	 * 
-	 * @param	string		$sessionID
+	 * Tries to read existing session identified by the given session id. Returns whether
+	 * a session could be found.
 	 */
-	protected function getExistingSession($sessionID) {
-		$this->session = new $this->sessionClassName($sessionID);
-		if (!$this->session->sessionID) {
-			$this->session = null;
-			return;
+	protected function getExistingSession(string $sessionID): bool {
+		$sql = "SELECT	*
+			FROM	wcf".WCF_N."_".($this->isACP ? 'acp' : 'user')."_session
+			WHERE	sessionID = ?";
+		$statement = WCF::getDB()->prepareStatement($sql);
+		$statement->execute([
+			$sessionID
+		]);
+		$row = $statement->fetchSingleRow();
+		
+		if (!$row) {
+			return false;
+		}
+		
+		// Check whether the session technically already expired.
+		$lifetime = ($this->isACP ? self::ACP_SESSION_LIFETIME :
+			($row['userID'] ? self::USER_SESSION_LIFETIME :
+			self::GUEST_SESSION_LIFETIME));
```

Comment:
```suggestion
		$lifetime = $row['userID'] ? self::USER_SESSION_LIFETIME : self::GUEST_SESSION_LIFETIME;
		if ($this->isACP) $lifetime = self::ACP_SESSION_LIFETIME;
```
This is only a suggestion, not an actual issue. I usually try to avoid nested shorthand, because I feel they are easily misinterpreted at a first glance.

Reply:
I agree that the nested if-expression is not super readable. I find your suggestion harder to understand, because the logic  is reversed. The winning case (ACP) comes last.

I just tried a full-blown if-else if-else, but that's super bloated. A dedicated method *could* work, but that requires the reader to jump around in the file. I'm trying to keep this code simple and straight forward for auditing reasons.

I've now committed a somewhat unorthodox suggestion with regard to the code style. But I find it very readable. What do you think?